File: scripts/osint.py
import requests
import sys

def get_subdomains(domain):
    url = f"https://crt.sh/?q=%25.{domain}&output=json"
    try:
        response = requests.get(url, timeout=10)
        if response.status_code != 200:
            print(f"Error al consultar crt.sh: {response.status_code}")
            return

        entries = response.json()
        subdomains = set()
        for entry in entries:
            name = entry.get("name_value")
            if name:
                subdomains.update(name.split("\n"))

        print(f"Subdominios encontrados para {domain}:")
        for sub in sorted(subdomains):
            print(f" - {sub}")

    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Uso: python osint.py <dominio>")
    else:
        get_subdomains(sys.argv[1])

File: cmd/osint.go
package main

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
	"os"
	"sort"
	"strings"
)

func main() {
	fmt.Println("\n[🕵️] Módulo OSINT activo")

	fmt.Print("[+] Ingresa el dominio objetivo (ejemplo.com): ")
	var target string
	fmt.Scanln(&target)

	if target == "" {
		fmt.Println("❌ Dominio no proporcionado.")
		return
	}

	url := fmt.Sprintf("https://crt.sh/?q=%%25.%s&output=json", target)
	resp, err := http.Get(url)
	if err != nil {
		fmt.Printf("❌ Error al consultar crt.sh: %v\n", err)
		return
	}
	defer resp.Body.Close()

	if resp.StatusCode != 200 {
		fmt.Printf("❌ Error HTTP: Código %d\n", resp.StatusCode)
		return
	}

	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		fmt.Printf("❌ Error al leer respuesta: %v\n", err)
		return
	}

	var data []map[string]interface{}
	err = json.Unmarshal(body, &data)
	if err != nil {
		fmt.Printf("❌ Error al parsear JSON: %v\n", err)
		return
	}

	subdomains := make(map[string]bool)

	for _, entry := range data {
		if nameValue, ok := entry["name_value"].(string); ok {
			names := strings.Split(nameValue, "\n")
			for _, name := range names {
				subdomains[name] = true
			}
		}
	}

	if len(subdomains) == 0 {
		fmt.Println("⚠️ No se encontraron subdominios.")
		return
	}

	fmt.Printf("[+] Subdominios encontrados para %s:\n", target)
	
	// Ordenar alfabéticamente antes de imprimir
	keys := make([]string, 0, len(subdomains))
	for sub := range subdomains {
		keys = append(keys, sub)
	}
	sort.Strings(keys)

	for _, sub := range keys {
		fmt.Printf(" - %s\n", sub)
	}

	fmt.Println("\n✅ Finalizado módulo OSINT.\n")
}
